mesheryctl/cmd/update.go

```diff
@@ -30,6 +37,74 @@ var updateCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		log.Info("Updating Meshery now...")
 
+		//Fetch latest version binary of mesheryctl
+		resp, err := http.Get(mesheryURL)
+		downloadMesheryURL := downloadMesheryURL
+		if err != nil {
+			log.Print("error")
+		} else {
+			var dat map[string]interface{}
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if err := json.Unmarshal(body, &dat); err != nil {
+				log.Fatal(err)
+			}
+			num := dat["tag_name"]
+			os := ""
+			arch := ""
+
+			if runtime.GOOS == "windows" {
+				// Meshery running on Windows host
+				os = "Windows"
+			} else if runtime.GOOS == "linux" {
+				// Meshery running on Linux host
+				os = "Linux"
+			} else {
+				// Assume Meshery running on MacOS host
+				os = "Darwin"
+			}
+
+			if runtime.GOARCH == "amd64" {
+				// Meshery running on x86_64 host
+				arch = "x86_64"
+			} else {
+				// Meshery running on i386 host
+				arch = "i386"
+			}
+
+			version := fmt.Sprint("%v", num)
+			downloadMesheryURL = fmt.Sprintf(downloadMesheryURL+"/%v/mesheryctl_%v_%v_%v.zip", num, version[3:], os, arch)
+		}
+
+		//If sudo is needed to update mesheryctl
+		user, err := user.Current()
+		if err != nil {
+			panic(err)
+		}
+		if user.Username == "root" {
+			if _, err := os.Stat(mesheryFolder); os.IsNotExist(err) {
+				_ = os.Mkdir(mesheryFolder, 0777)
+			}
+		}
+
+		//download mehseryctl binary
+		if err := downloadFile(mesherybinary, downloadMesheryURL); err != nil {
+			log.Fatal("update cmd: ", err)
+		}
+
+		//Unzip mesheryctl binary
+		uz := unzip.New(mesherybinary, "/usr/local/bin/")
+		err = uz.Extract()
+		if err != nil {
+			log.Warn(err)
+			log.Warn("skiping mesheryctl binary update!")
```

Comment:
"skiping" is misspelled.
If we're skipping due to a permission error, let's include this warning. Otherwise, if we're skipping because the version # is the same as what the user already has on their system, then no need for a Warning, just an informational log will suffice.

Let's check to see if the current `mesheryctl` is the same as the remote (latest) `mesheryctl` version and provide an informational log stating that we're skipping, because the user already has the latest `mesheryctl`.

In order to handle the issue of lack of permissions, let's prompt the user at this point for their sudo passwd and re-execute this unzip under that privileged context.